### PR TITLE
Improve global UX

### PR DESCRIPTION
**Pull request by stafyniaksacha**
Some preview:

http://www.awesomescreenshot.com/image/1499319/98095e753ae3b97ba2eab45e35a0d69b
http://www.awesomescreenshot.com/image/1499327/38cfe7c69c8d1bbf769c8cf46202fc33
http://www.awesomescreenshot.com/image/1499329/9dd567941f0aab5e082b66f7d447eaff
http://www.awesomescreenshot.com/image/1499334/14469f6341aa3f330441affb67d403df
http://www.awesomescreenshot.com/image/1499336/beacc020b4cc972b4dd0e968a66e0fdc

Todo: 
- [x] browse search bar
- [x] all security part
- [x] delete collection toggle autorefresh
- [x] index & collection view when empty result on filtering
- [x] improve filters
- [x] improve edit profile / roles
- [x] more rights management
  - [x] watch
  - [x] browse collection
  - [x] browse index
  - ~~browse documents~~ **(must be done in other PR KUZBO-148)**
  - ~~browse security (Role/Profile/User)~~ **(must be done in other PR KUZBO-149)**

**Note:** The "CREATE A DOCUMENT" tab is still available from a realtime collection because I'm waiting for @jenow 's PR.

### CODE REVIEW

**Comment by stafyniaksacha** [comment]
Small improvement: 

On the "browse" page of a realtime collection, there is a small improvement to do:
http://www.awesomescreenshot.com/image/1510479/0a53be12d20f71d76bde180071c3d56e

- Whether we let the "browse" tab on the realtime collection and we had to change the message & the label button saying that user have to create a document to persist the collection (the functionality is working) 
- Or hide the tab (but user will have to edit the collection to persist it)

**Comment by xbill82** [comment]
@stafyniaksacha I don't think we should _hide_ the "Browse" tab, since it would decrease discoverability. I wouldn't even make it grey. I would leave it as-is and let the user enter the tab and show him a clear message saying something like

> There are no presisted documents in here because this collection is currently realtime-only ([what does this mean?](http://www.tell-the-user.com/)). You can **click here** to persist it or **click here** to create a document (the collection will be automatically persisted).

This way, the user will have clearly explained and easy-to-access call-to-actions.

**Comment by AnthonySendra** [comment]
@xbill82 I don't want to let the user create a document without be totally aware of mapping, because after that, it's to late... 
I prefer to let only the button for redirecting the user on the edit collection page.

> There are no presisted documents in here because this collection is currently realtime-only [(what does this mean?](http://www.tell-the-user.com/)). You can **click here** to edit the collection and persist it.

**Comment by AnthonySendra** [comment]
*Note:* The "create document" tab is still available from a realtime collection because I'm waiting for @jenow 's PR.

**Comment by xbill82** [comment]
Wording problem in collections empty view
http://www.awesomescreenshot.com/showImage?img_id=1510884

File: src/components/Data/Indexes/List.vue [hunk 1]
```diff
@@ -37,23 +37,38 @@
...
-          <div class="col s12" v-if="canSearchIndex() && !indexesAndCollections">
-            <a class="card-title"
+          <div class="col s12" v-if="canSearchIndex() && !indexesAndCollections.length">
+            <a class="card-title fluid-hover"
              href="#!"
              @click.prevent="$broadcast('modal-open', 'index-create')">
               <div class="card-panel hoverable">
                 <div class="card-content">
-                    There is no index yet. You may want to create a new one ?
+                    <em>There is no index yet. You may want to create a new one ?</em>
```
Comment (xbill82) [inline src/components/Data/Indexes/List.vue]: Wording problem:

> There is currently no index in your database. You may want to create one and start working.

File: src/components/Data/Indexes/List.vue [hunk 2]
```diff
@@ -37,23 +37,38 @@
...
+          <!-- No index for filter -->
+          <div class="card-panel card-body" v-if="!countIndexForFilter && filter">
+            <div class="row valign-center empty-set">
+              <div class="col s2 offset-s1">
+                <i class="fa fa-6x fa-search grey-text text-lighten-1" aria-hidden="true"></i>
+              </div>
+              <div class="col s12">
+                <p>
+                  There is no index matching your filter.<br />
+                  Please try with other filter.
```
Comment (xbill82) [inline src/components/Data/Indexes/List.vue]: Wording problem

> Please try with another filter.

File: src/components/Data/Indexes/List.vue [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
-             :class="{unauthorized: !canCreateIndex()}"
-             @click.prevent="canCreateIndex() && $broadcast('modal-open', 'index-create')">
-            <i class="fa fa-plus-circle left"></i>
-            <span>Create</span>
-          </a>
-        </div>
+        <div class="row actions">
+          <div class="col s9">
+            <a class="btn waves-effect waves-light primary"
+               v-title="{active: !canCreateIndex(), title: 'Your rights disallow you to create indexes'}"
```
Comment (xbill82) [inline src/components/Data/Indexes/List.vue]: Wording

> You are not allowed to create new indexes.

File: src/components/Common/CrudlDocument.vue [hunk 4]
```diff
@@ -16,89 +16,101 @@
...
-              <pagination
+    <div class="card-panel card-body">
+      <div class="row valign-center empty-set" v-show="!documents.length">
+        <div class="col s2 offset-s1">
+          <i class="fa fa-6x fa-search grey-text text-lighten-1" aria-hidden="true"></i>
+        </div>
+        <div class="col s12">
+          <p>
+            There is no result matching your query<br />
+            Please try with other filters.
```
Comment (xbill82) [inline src/components/Common/CrudlDocument.vue]: > Please try with another filter.

File: src/components/Common/CrudlDocument.vue [hunk 5]
```diff
@@ -16,89 +16,101 @@
...
+                I'm sure!
+            </button>
+            <button href="#" class="btn-flat" @click.prevent="$broadcast('modal-close', 'bulk-delete')">
+                Cancel
+            </button>
+          </span>
     </modal>
 
     <modal id="single-delete">
       <h4>Users deletion</h4>
```
Comment (xbill82) [inline src/components/Common/CrudlDocument.vue]: > Delete user

File: src/components/Data/Collections/Watch.vue [hunk 6]
```diff
@@ -37,50 +20,115 @@
...
+        </div>
+
+        <div class="card-panel card-body" v-show="canSubscribe(index, collection) && !subscribed && !notifications.length">
+          <div class="row valign-bottom empty-set">
+            <div class="col s1 offset-s1">
+              <i class="fa fa-6x fa-paper-plane grey-text text-lighten-1" aria-hidden="true"></i>
+            </div>
+            <div class="col s10">
+              <p>
+                You did not subscribed yet to the collection <strong>{{collection}}</strong><br>
```
Comment (xbill82) [inline src/components/Data/Collections/Watch.vue]: > You did not subscribe yet to the collection...

File: src/components/Data/Collections/Watch.vue [hunk 7]
```diff
@@ -37,50 +20,115 @@
...
+        </div>
+
+        <div class="card-panel" v-show="canSubscribe(index, collection) && subscribed && !notifications.length">
+          <div class="row valign-center empty-set empty-set-condensed">
+            <div class="col s1 offset-s1">
+              <i class="fa fa-5x fa-hourglass-half grey-text text-lighten-1" aria-hidden="true"></i>
+            </div>
+            <div class="col s10">
+              <p>
+                Waiting for notification matching your filters ...
```
Comment (xbill82) [inline src/components/Data/Collections/Watch.vue]: moi je mettrais "notifications" au pluriel

File: src/components/Data/Documents/List.vue [hunk 8]
```diff
@@ -1,24 +1,65 @@
...
       @create-clicked="createDocument">
 
+      <div slot="emptySet" class="card-panel">
+        <div v-if="isRealtimeCollection" class="row valign-bottom empty-set">
+          <div class="col s1 offset-s1">
+            <i class="fa fa-6x fa-file-text-o grey-text text-lighten-1" aria-hidden="true"></i>
+          </div>
+          <div class="col s10">
+            <p>
+              There are no presisted document in here because the collection <strong>{{collection}}</strong> is currently realtime-only.<br />
```
Comment (xbill82) [inline src/components/Data/Documents/List.vue]: > There is no persistent document

File: src/components/Data/Documents/List.vue [hunk 9]
```diff
@@ -1,24 +1,65 @@
...
+          </div>
+        </div>
+
+        <div v-else class="row valign-bottom empty-set">
+          <div class="col s1 offset-s1">
+            <i class="fa fa-6x fa-file-text-o grey-text text-lighten-1" aria-hidden="true"></i>
+          </div>
+          <div class="col s10">
+            <p>
+              You will see <strong>{{collection}}'s</strong> documents here<br/>
```
Comment (xbill82) [inline src/components/Data/Documents/List.vue]: > Here you'll see the documents in ${collection_name} 
> Currently there is no document in this collection.

**Comment by xbill82** [comment]
:+1:

**Comment by jenow** [comment]
+1

**Pull request merged**